### PR TITLE
Collapse repeated

### LINKED ISSUE #12491: collapse_repeated

### PR DESCRIPTION
**Pull request by AdityaVadlamani**
Close #12491

### CODE REVIEW

**Comment by krishna1111111** [comment]
Once check with the ivy API experimental and array api

**Comment by AdityaVadlamani** [comment]
I can check the functionality using both APIs. Is there a reference in the docs on how to do so, or is it just installing ivy-core using pip?

**Comment by AdityaVadlamani** [comment]
I went ahead and tested it with the Ivy API and the array API with the following script:

```
import jax.numpy as jnp
import tensorflow as tf
import numpy as np
import torch

import ivy

ivy.set_backend("jax")
jax_collapse_repeated = ivy.collapse_repeated(jnp.array([[0, 0, 1, 1, 0], [1, 1, 2, 3, 4]]), 5)

ivy.set_backend("tensorflow")
tf_collapse_repeated = ivy.collapse_repeated(tf.convert_to_tensor([[0, 0, 1, 1, 0], [1, 1, 2, 3, 4]]), 5)

ivy.set_backend("numpy")
np_collapse_repeated = ivy.collapse_repeated(np.array([[0, 0, 1, 1, 0], [1, 1, 2, 3, 4]]), 5)

ivy.set_backend("torch")
torch_collapse_repeated = ivy.collapse_repeated(torch.as_tensor([[0, 0, 1, 1, 0], [1, 1, 2, 3, 4]]), 5)

ivy_collapse_repeated = ivy.array([[0, 0, 1, 1, 0], [1, 1, 2, 3, 4]]).collapse_repeated(5)

print(jax_collapse_repeated)
print(tf_collapse_repeated)
print(np_collapse_repeated)
print(torch_collapse_repeated)
print(ivy_collapse_repeated)
```
I got the expected results for each of the backends. Let me know if there's something else I need to test!

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi. Thanks for the this amazing PR. 
You have implemented this compositionally which is totally fine. However, Tensorflow does support collapse_repeated function and we should try to implement the tensorflow backend `collapse_repeated` using the call to tensorflow. This ensures effieciency.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
